[Issue 5401][docs] Add deploy-docker doc (#7769)
apache/pulsar: site2/docs/deploy-docker.md

```

### Create three containers
Create containers for ZooKeeper, BookKeeper and broker. In this example, they are named as `zookeeper`, `bookkeeper` and `broker` respectively. You can name them as you want with the `--name` flag. By default, the container names are created randomly.

```
docker run -it --name bookkeeper apachepulsar/pulsar-all:latest /bin/bash
docker run -it --name zookeeper apachepulsar/pulsar-all:latest /bin/bash
docker run -it --name broker apachepulsar/pulsar-all:latest /bin/bash
```

### Create a network
To deploy a Pulsar cluster on Docker, you need to create a `network` and connect the containers of ZooKeeper, BookKeeper and broker to this network. The following command creates the network `pulsar`:

```
docker network create pulsar
```

### Connect containers to network
Connect the containers of ZooKeeper, BookKeeper and broker to the `pulsar` network with the following commands. 

```
docker network connect pulsar zookeeper
docker network connect pulsar bookkeeper
docker network connect pulsar broker
```

To check whether the containers are successfully connected to the network, enter the `docker network inspect pulsar` command.

> For detailed information about deployment for each Pulsar component, see [deploy a cluster on bare metal](https://pulsar.apache.org/docs/en/deploy-bare-metal/).

Reviewer: ```suggestion
For detailed information about how to deploy ZooKeeper cluster, BookKeeper cluster, brokers, see [deploy a cluster on bare metal](deploy-bare-metal.md).
```

Reviewer: 1. When you tell users Pulsar component, they might think which component.
2. Use relative path in reference link.
